pkg/build/apis/build/types.go

```diff
@@ -1054,11 +1054,16 @@ type BuildConfigStatus struct {
 // WebHookTrigger is a trigger that gets invoked using a webhook type of post
 type WebHookTrigger struct {
 	// Secret used to validate requests.
+	// Deprecated: use Secret instead.
 	Secret string
 
 	// AllowEnv determines whether the webhook can set environment variables; can only
 	// be set to true for GenericWebHook
 	AllowEnv bool
+
+	// SecretRef is a reference to a secret containing the value to be validated
+	// when the webhook is invoked.  Takes precedence over Secret.
+	SecretRef *kapi.LocalObjectReference
```

Comment:
Upstream we've started defining specific types for each kind of reference.  The idea being that this isn't a generic reference, this is a reference to a secret and even more specifically, a reference to a particularly kind of secret.  Basically, every reference started looking a little different and chasing them all around is impractical.

Reply:
I defined a new type for the reference.  I did not add (and don't plan to add?) a new type of secret.